Marlin/src/lcd/extui/ui_api.cpp

```diff
@@ -982,21 +982,23 @@ namespace ExtUI {
       default: {
         #if HAS_HOTEND
           const int16_t e = heater - H0;
-          thermalManager.setTargetHotend(LROUND(constrain(value, 0, thermalManager.hotend_max_target(e))), e);
+          LIMIT(value, 0, thermalManager.hotend_max_target(e));
+          thermalManager.setTargetHotend(LROUND(float(value)), e);
         #endif
       } break;
     }
   }
 
   void setTargetTemp_celsius(const_float_t inval, const extruder_t extruder) {
```

Comment:
By using `float` here, ExtUI is likely keeping the CocoaPress chocolate printer in mind, because it needs to be able to set temperatures in fractions of degrees.

Reply:
I don't see any issue doing some computation in float where it's required. In my next changes I'll put a comment about this reason here.